src/communication/communicator.hpp

```diff
@@ -101,17 +102,54 @@ class communicator {
     ///
     /// Returns a vector of event queues, with one queue for each local cell group. The
     /// events in each queue are all events that must be delivered to targets in that cell
-    /// group as a result of the global spike exchange.
-    std::vector<event_queue> make_event_queues(const gathered_vector<spike_type>& global_spikes) {
-        auto queues = std::vector<event_queue>(num_groups_local());
-        for (auto spike : global_spikes.values()) {
-            // search for targets
-            auto targets = std::equal_range(connections_.begin(), connections_.end(), spike.source);
-
-            // generate an event for each target
-            for (auto it=targets.first; it!=targets.second; ++it) {
-                auto gidx = cell_group_index(it->destination().gid);
-                queues[gidx].push_back(it->make_event(spike));
+    /// group as a result of the global spike exchange.    
+    std::vector<event_queue> make_event_queues(const gathered_vector<spike_type>& global_spikes)
+    {
+        // Comparator operator between a spike and a spike source for equal_range
+        struct comp {
+            using id_type = typename spike_type::id_type;
+            bool operator() (const spike_type& lhs, const id_type& rhs) const
+            {return lhs.source < rhs;}
```

Comment:
Well, it is one extra line compared to your version. And we have been through this before. You get a lot of noise if it is all on one line and the reader has to parse it.